### PR TITLE
test lc expiration with dynamic resharding

### PR DESCRIPTION
**Pull request by hmaheswa**
this PR is to integrate lifecycle and dynamic resharding features.
polarion: https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-83574679

and to verify attr after 20 mins,  refer [Bz 2037330#c17](https://bugzilla.redhat.com/show_bug.cgi?id=2037330#c17)

pass logs:
http://magna002.ceph.redhat.com/ceph-qe-logs/HemanthSai/lc-dynamic-reshard-PR-logs/

### CODE REVIEW

File: rgw/v2/tests/s3_swift/reusable.py [hunk 1]
```diff
@@ -1419,8 +1419,86 @@ def bucket_reshard_manual(bucket, config):
...
+        log.info("no_of_shards_expected: %s" % num_shards_expected)
+        if int(num_shards_created) >= int(num_shards_expected):
+            log.info("Expected number of shards created")
+    else:
+        raise TestExecError("Expected number of shards not created")
+    verify_attrs_after_resharding(bucket, bucket_id)
+
+
+def verify_attrs_after_resharding(bucket, bucket_id):
+    log.info("Test acls are preserved after a resharding operation.")
```
Comment (viduship) [inline rgw/v2/tests/s3_swift/reusable.py:1495]: @hmaheswa , this is a good test to include. But in order to test the acl preservation, (you may only do it for one test only), please give a sleep interval of 20mins , sometimes post which the acls do disppear.

ref : https://bugzilla.redhat.com/show_bug.cgi?id=2037330#c17 and https://bugzilla.redhat.com/show_bug.cgi?id=2037330#c19

**Review by viduship — CHANGES_REQUESTED** [review]

File: rgw/v2/tests/s3_swift/reusable.py (hunk 1 above)
Comment (hmaheswa) [reply rgw/v2/tests/s3_swift/reusable.py:1494]: @viduship , addressed the comment, please review again

File: rgw/v2/tests/s3_swift/reusable.py [hunk 2]
```diff
@@ -1424,3 +1424,83 @@ def bucket_reshard_manual(bucket, config):
...
+    num_shards_created = json_doc["num_shards"]
+    log.info("no_of_shards_created: %s" % num_shards_created)
+    log.info("Verify if resharding list is empty")
+    reshard_list_op = json.loads(utils.exec_shell_cmd("radosgw-admin reshard list"))
+    if not reshard_list_op:
+        log.info(
+            "for dynamic number of shards created should be greater than or equal to number of expected shards"
+        )
+        log.info("no_of_shards_expected: %s" % num_shards_expected)
+        if int(num_shards_created) >= int(num_shards_expected):
```
Comment (ckulal) [inline rgw/v2/tests/s3_swift/reusable.py:1486]: If num_shards_created < num_shards_expected we need to raise assertion ryt?

File: rgw/v2/tests/s3_swift/reusable.py (hunk 1 above)
Comment (viduship) [reply rgw/v2/tests/s3_swift/reusable.py:1502]: Looks good.

File: rgw/v2/tests/s3_swift/reusable.py (hunk 2 above)
Comment (hmaheswa) [reply rgw/v2/tests/s3_swift/reusable.py:1494]: @ckulal , addressed the comment

**Review by viduship — APPROVED** [review]

**Review by anrao19 — APPROVED** [review]

**Review by ckulal — COMMENTED** [review]
Looks good

**Review by mkasturi18 — APPROVED** [review]

**Pull request merged**